internal/verifier/migration_verifier.go

```diff
@@ -875,64 +866,74 @@ func (verifier *Verifier) compareCollectionSpecifications(srcNs string, dstNs st
 	return results, canCompareData
 }
 
-func compareIndexSpecifications(srcSpec *mongo.IndexSpecification, dstSpec *mongo.IndexSpecification) []VerificationResult {
-	var results []VerificationResult
-	// Order is always significant in the keys document.
-	if !bytes.Equal(srcSpec.KeysDocument, dstSpec.KeysDocument) {
-		results = append(results, VerificationResult{
-			NameSpace: dstSpec.Namespace,
-			Cluster:   ClusterTarget,
-			ID:        dstSpec.Name,
-			Field:     "KeysDocument",
-			Details:   Mismatch + fmt.Sprintf(" : src: %v, dst: %v", srcSpec.KeysDocument, dstSpec.KeysDocument)})
+func (verifier *Verifier) doIndexSpecsMatch(ctx context.Context, srcSpec bson.Raw, dstSpec bson.Raw) (bool, error) {
+	// If the byte buffers match, then we’re done.
+	if bytes.Equal(srcSpec, dstSpec) {
+		return true, nil
 	}
 
-	// We don't check version because it may change when migrating between server versions.
-
-	if !reflect.DeepEqual(srcSpec.ExpireAfterSeconds, dstSpec.ExpireAfterSeconds) {
-		results = append(results, VerificationResult{
-			NameSpace: dstSpec.Namespace,
-			Cluster:   ClusterTarget,
-			ID:        dstSpec.Name,
-			Field:     "ExpireAfterSeconds",
-			Details:   Mismatch + fmt.Sprintf(" : src: %s, dst: %s", nilableToString(srcSpec.ExpireAfterSeconds), nilableToString(dstSpec.ExpireAfterSeconds))})
+	// Next check to see if the only differences are type differences.
+	// If we didn’t support pre-v5 servers we could use a $documents aggregation,
```

Comment:
I think we can reasonably limit the metadata cluster to post-v5 versions, but probably doesn't worth to do only for `$document`.

Reply:
ahhh, right! I’d overlooked that this is the metadata cluster. Totally worth it, then, IMO.